src/components/debounced-input.js

```diff
@@ -16,18 +16,32 @@ function DebouncedInput({ elementType: Component, value, onChange, ...props }) {
     debouncedOnChange(event.target.value);
   };
 
-  return <Component {...props} value={actualValue} onChange={handleChange} />;
+  const componentProps = { ...props };
+
+  if (onEnter) {
+    componentProps.onEnter = () => onEnter(actualValue);
+  }
+
+  if (onSearch) {
+    componentProps.onSearch = () => onSearch(actualValue);
+  }
+
+  return <Component {...componentProps} value={actualValue} onChange={handleChange} />;
```

Comment:
Although this works in our scenarios, we should keep in mind that it's not a "multi-purpose" implementation, meaning that if it's used as a controlled component, the `value` value that's kept in state outside has still the old value and might be used. The expected behavior would be - in the event the `Component` fires `onEnter` or `onSearch` - to first cancel the debounce and trigger a regular `onChange`, so the outside world is informed and `value` (as seen from the outside) is updated, before triggering `onEnter` or `onSearch`. But that would either require us to add "force-updating" to the debounce hook or manually trigger the `onChange` callback first in this component (maybe in combination with `useDedupedCallback` so we don't call it again with the same value when the actual debounced value change kicks in), before any `onEnter` or `onSearch`. What do you think about leaving it for the purpose of this ticket and having an extra ticket for this? Even though it might not impact the way we use the component right now, we would prevent some possible surprises in the future.

Reply:
That is true, that having `onSearch`/`onEnter` carry the updated value breaks the flow of using `onChange` for this purpose. Indeed we should first call `onChange`.
I can revert this change and make a ticket for this purpose then.
Thanks for pointing this out!

Reply:
You could also leave it as a temporary fix if we're gonna change it soon anyways, it's up to you. Extra ticket sounds good to me.

Reply:
Ok, will not revert then so that we at least have the fix for now.
Will describe the changes needed in the follow-up ticket https://educandu.atlassian.net/browse/EDU-811

Reply:
:+1: